frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js

```diff
@@ -585,29 +721,81 @@ export default function WaveformGraph({
       .append("line")
       .attr("id", "peakLine")
       .attr("x1", x(startTime))
-      .attr("y1", y(minPeaks))
+      .attr("y1", y(peakY1[wellNameToIndex[currentWell]]))
       .attr("x2", x(endTime))
-      .attr("y2", y(minPeaks))
+      .attr("y2", y(peakY2[wellNameToIndex[currentWell]]))
       .attr("stroke-width", 2)
       .attr("stroke", "orange")
       .style("cursor", "pointer")
-      .call(peaksValleysLineDrag);
+      .call(moveLineUpDown);
+    const peaksY1 = svg
+      .append("circle")
+      .attr("id", "peakLineY1Marker")
+      .attr("lineId", "peakLine")
+      .attr("r", 6)
+      .attr("cx", x((endTime - startTime) / 100))
+      .attr("cy", y(peakY1[wellNameToIndex[currentWell]]))
+      .attr("fill", "orange")
+      .attr("stroke", "orange")
+      .style("cursor", "pointer")
+      .call(pivotLineDrag);
+    const peaksY2 = svg
```

Comment:
I think we could make a function for this as well

Reply:
A function to create each draggable point at the ends of each line?

Reply:
Whatever this is doing. There is a lot of duplicate code here creating `peaksY1` and `peaksY2`. I think the same is true for the valley equivalents